mvpa2/misc/plot/scatter.py

```diff
@@ -0,0 +1,643 @@
+      is quantized so it is valuable to jitter points a bit.
+    y_jitter: float, optional
+      Half-width of uniform noise added to y values.  Might be useful if data
+      is quantized so it is valuable to jitter points a bit
+    fig : Figure, optional
+      Figure to plot on, otherwise new one created
+    ax_*: axes, optional
+      Axes for the scatter plot and histograms. If none of them is specified
+      (which is the default) then 'classical' plot is rendered with histograms
+      above and to the right
+    bp_location: ('scatter', 'hist', None), optional
+      Where to place boxplots depicting data range
+    xlim: tuple, optional
+    ylim: tuple, optional
+      To fix plotted range
+    rasterized: bool, optional
+      Passed to scatter call, to allow rasterization of heavy scatter plots
+    uniq: bool, optional
+      Plot uniq values (those present in one but not in the other) along each axis
+      with crosses
+    """
+    if len(dataXd) != 2:
+        raise ValueError("First axis of dataXd can only have two dimensions, "
+                         "got {0}".format(len(dataXd)))
+    dataXd = np.asanyarray(dataXd)      # TODO: allow to operate on list of arrays to not waste RAM/cycles
+    data = dataXd.reshape((2, -1))
+    if dataXd.ndim < 5:
+        ntimepoints = 1
+    elif dataXd.ndim == 5:
+        ntimepoints = dataXd.shape[-1]
+    else:
+        raise ValueError("Do not know how to handle data with %d dimensions" % (dataXd.ndim - 1))
+    if x_jitter or y_jitter:
+        data = data.copy()              # lazy and wasteful
+        def jitter_me(x, w):
+            x += np.random.uniform(-w, w, size=data.shape[-1])
+        if x_jitter:
+            jitter_me(data[0, :], x_jitter)
+        if y_jitter:
+            jitter_me(data[1, :], y_jitter)
+
+    finites = np.isfinite(data)
+    nz = np.logical_and(data != 0, finites)
+    # TODO : avoid doing data !=0 and just use provided utter mask
+    #nz[:, 80000:] = False # for quick testing
+
+    nzsum = np.sum(nz, axis=0)
+
+    intersection = nzsum == 2
+    # for coloring we would need to know all the indices
+    union = nzsum > 0
+    x, y = datainter = data[:, intersection]
+
+    if mask is not None:
+        # replicate mask ntimepoints times
+        mask = np.repeat(mask.ravel(), ntimepoints)[intersection] != 0
+
+    xnoty = (nz[0].astype(int) - nz[1].astype(int))>0
+    ynotx = (nz[1].astype(int) - nz[0].astype(int))>0
+
+    msg = ''
+    if not np.all(finites):
+        msg = " non-finite x: %d, y: %d" % (np.sum(~finites[0]), np.sum(~finites[1]))
+
+    verbose(1, "total: %d union: %d%s intersection: %d x_only: %d y_only: %d%s"
+            % (len(nzsum),
+               np.sum(union),
+               mask is not None and ' masked: %d' % np.sum(mask) or '',
+               np.sum(intersection),
+               np.sum(xnoty), np.sum(ynotx),
+               msg))
+
+    if verbose.level > 1:
```

Comment:
please add a 'TODO:  make this one independent of verbose.level - intoduce some argument/handling' line